core-v1/src/main/scala/dsl/BuildExprDsl.scala

```diff
@@ -28,6 +29,16 @@ trait BuildExprDsl extends DebugExprDsl {
 
   def valuesOfType[T](factTypeSet: FactTypeSet[T])(implicit opTs: OP[Seq[W[T]]]): Expr.ValuesOfType[T, W[T], OP]
 
+  final def pow[I, L, R](
+    leftExpr: I ~:> W[L],
+    rightExpr: I ~:> W[R],
+  )(implicit
+    opR: OP[W[R]],
+    pow: Power[W[L], W[R]],
+  ): CombineHolder[I, W[L], W[L], W[R], W[R], pow.Out, OP] = {
+    (leftExpr ^ rightExpr)(opR, pow)
+  }
```

Comment:
I should let DSL subclasses define this to avoid unnecessary wrapper types hanging around in the unwrapped case.